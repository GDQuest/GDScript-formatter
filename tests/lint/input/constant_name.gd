#region Good

const GOOD_NAME = 42
const _PRIVATE_GOOD_NAME = 100

# preload can be CONSTANT_CASE or PascalCase (i.e. a ClassName)
const GOOD_LOAD = preload("res://path/to/resource.tscn")
const ClassNameLoad = preload("res://path/to/resource2.tscn")

#endregion

#region Bad

const bad_name = 42
const _private_bad_name = 43
const AnotherBadName = 44

const bad_preload = preload("res://path/to/resource3.tscn")

# this is technically invalid GDScript since `load` can't be assigned to a constant
const BAD_LOAD = load("res://path/to/resource4.tscn")

#endregion