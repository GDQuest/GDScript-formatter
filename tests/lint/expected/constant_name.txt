tests/lint/input/constant_name.gd:14:constant-name:error: Constant name 'bad_name' should be in CONSTANT_CASE format
tests/lint/input/constant_name.gd:15:constant-name:error: Constant name '_private_bad_name' should be in CONSTANT_CASE format
tests/lint/input/constant_name.gd:16:constant-name:error: Constant name 'AnotherBadName' should be in CONSTANT_CASE format
tests/lint/input/constant_name.gd:18:constant-name:error: Preload constant name 'bad_preload' should be in PascalCase or CONSTANT_CASE format